### PR TITLE
feat: allow configuring loadingStages and customizing reportFn

### PR DESCRIPTION
**Pull request by bbrzoska**
also fixes the issue of "$previousStage to $stage" being both reported as the same 🤞

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -132,12 +139,24 @@ export interface WithMetadata<Metadata extends Record<string, unknown>> {
...
+  readonly actions: readonly ActionWithStateMetadata[]
+  readonly timingId?: string
+  readonly isFirstLoad?: boolean
+  readonly maximumActiveBeaconsCount?: number
+  readonly minimumExpectedSimultaneousBeacons?: number
+  readonly flushReason?: string
+  readonly metadata?: CustomMetadata
 }
 
+// TODO remove
```
Comment (xnanodax) [inline src/types.ts:156]: did we still want to remove this?

File: README.md [hunk 2]
```diff
@@ -32,7 +32,10 @@ First, create a file that generates and exports the hooks for your metric ID pre
 For example:
 
 ```typescript
-import { generateTimingHooks } from '@zendesk/react-measure-timing-hooks'
+import {
+  generateTimingHooks,
+  defaultReport,
```
Comment (xnanodax) [inline README.md:37]:
```suggestion
  generateReport,
```

File: src/ActionLog.ts [hunk 3]
```diff
@@ -410,16 +414,14 @@ export class ActionLog<CustomMetadata extends Record<string, unknown>> {
...
 
-      const mark = performanceMark(`${this.id}/longTaskEnd`)
-
       const entries = entryList.getEntries()
 
       for (const entry of entries) {
         this.addSpan({
           type: ACTION_TYPE.UNRESPONSIVE,
           source: OBSERVER_SOURCE,
           // workaround for the fact that long-task performance measures do not have unique names:
```
Comment (tenshiemi) [inline src/ActionLog.ts:423]: Can this comment be removed now?

**Review by xnanodax — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bbrzoska) [reply README.md:37]: oops, search and replace strikes back!

File: src/types.ts (hunk 1 above)
Comment (bbrzoska) [reply src/types.ts:156]: yup!

**Pull request merged**